account_invoice_inter_company/models/account_invoice.py

```diff
@@ -173,13 +173,20 @@ def _prepare_invoice_data(self, dest_inv_type, dest_journal_type,
         }
         dest_partner_data = self.env['account.invoice'].play_onchanges(
             dest_partner_data, ['partner_id'])
+        if not dest_partner_data.get('payment_term_id'):
+            # check access rule for payment term
+            domain = self._get_user_domain(dest_company)
+            dest_user = self.env['res.users'].search(domain, limit=1)
+            if not self.payment_term_id.sudo(
+                    dest_user).check_access_rule('read'):
+                dest_partner_data['payment_term_id'] = self.payment_term_id.id
         return {
             'name': self.name,
             'origin': _('%s - Invoice: %s') % (self.company_id.name,
                                                self.number),
             'type': dest_inv_type,
             'date_invoice': self.date_invoice,
-            'reference': self.reference,
+            'reference': self.number,
```

Comment:
Why this change?

Reply:
@pedrobaeza ,I think it's clearer to have the original invoice number in the 'supplier reference' field.
I know it can duplicate with the field 'origin'.
By the way, the field 'origin', I could leave it empty.

Reply:
OK, thanks for the clarification. Indeed it's a good option. Leave both.